sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala

```diff
@@ -372,17 +492,82 @@ class WriteDistributionAndOrderingSuite
         Seq.empty
       )
     )
-    val writePartitioning = RangePartitioning(writeOrdering, conf.numShufflePartitions)
+    val writePartitioning = orderedWritePartitioning(writeOrdering, targetNumPartitions)
 
     checkWriteRequirements(
       tableDistribution,
+      targetNumPartitions,
       tableOrdering,
       expectedWritePartitioning = writePartitioning,
       expectedWriteOrdering = writeOrdering,
       writeTransform = df => df.sortWithinPartitions("data", "id"),
       writeCommand = command)
   }
 
+  ignore("ordered distribution and sort with manual repartition: append") {
```

Comment:
Shall we add the JIRA here? Is SPARK-34184 the right one?

Reply:
Ah OK this PR was based on your PR which didn't remove ignored tests. Probably safer to remove here as well, with adding change in JIRA comment. Will do.

Reply:
Thanks!